BNomad/View/MapView/MapViewController.swift

```diff
@@ -374,6 +394,26 @@ extension MapViewController: MKMapViewDelegate {
         userAnnotationView?.isEnabled = false
     }
     
+
+    func mapViewDidChangeVisibleRegion(_ mapView: MKMapView) {
+        for place in viewModel.places {
+            if mapView.visibleMapRect.contains(MKMapRect(origin: MKMapPoint(CLLocationCoordinate2D(latitude: place.latitude, longitude: place.longitude)), size: MKMapSize(width: 0.2, height: 0.2))) // 맵 이동 시 정확도 검증을 위해 남겨둡니다.
+//            if mapView.visibleMapRect.contains(MKMapPoint(CLLocationCoordinate2D(latitude: place.latitude, longitude: place.longitude)))
+            {
+                print(place, "visible")
+                if visiblePlacesOnMap.contains { $0.name == place.name } {
+                    return
+                }
+                visiblePlacesOnMap.append(place)
+            } else {
+                print(place, "invisible")
+                visiblePlacesOnMap.removeAll { $0.name == place.name }
```

Comment:
```suggestion
    func mapViewDidChangeVisibleRegion(_ mapView: MKMapView) {
        visiblePlacesOnMap = []
        for place in viewModel.places {
            if mapView.visibleMapRect.contains(MKMapRect(origin: MKMapPoint(CLLocationCoordinate2D(latitude: place.latitude, longitude: place.longitude)), size: MKMapSize(width: 0.2, height: 0.2))) // 맵 이동 시 정확도 검증을 위해 남겨둡니다.
//            if mapView.visibleMapRect.contains(MKMapPoint(CLLocationCoordinate2D(latitude: place.latitude, longitude: place.longitude)))
            {
                print(place, "visible")
                visiblePlacesOnMap.append(place)
            } else {
                print(place, "invisible")
                visiblePlacesOnMap.removeAll { $0.name == place.name }
```

`contains` 로 하는 방법대신, `mapViewDidChangeVisibleRegion` 실행되고 가장먼저 `visiblePlacesOnMap`을 빈 배열로 만들어주면 기가막히게 잘 동작하네요..!!!!!👍 오우 씌잇,,, 죽이네여

`return`이 잘 잘동안했던 이유는 `return`이 for 구문으로 돌아가라는 명령어가 아니라, `mapViewDidChangeVisibleRegion`함수 자체를 `return` 했기 때문입니다😲

Reply:
크으.. 고맙습니다!!